app/models/host/managed.rb

```diff
@@ -918,8 +923,10 @@ def recreate_config
   # if we are managing DNS, we can query the correct DNS server
   # otherwise, use normal systems dns settings to resolv
   def to_ip_address(name_or_ip)
-    return name_or_ip if name_or_ip =~ Net::Validations::IP_REGEXP
-    if dns_ptr_record
+    ip4 = Net::Validations.validate_ip(name_or_ip)
+    ip6 = Net::Validations.validate_ip6(name_or_ip)
+    return name_or_ip if ip4 || ip6
+    if dns_ptr_record && ip4
       lookup = dns_ptr_record.dns_lookup(name_or_ip)
       return lookup.ip unless lookup.nil?
     end
```

Comment:
Isnt the block 930-931 unreachable? And why not to do DNS lookup for IPv6?

Hello, thanks for your contribution :-)